SU2_CFD/src/solvers/CPassiveScalarSolver.cpp

```diff
@@ -237,9 +241,49 @@ void CPassiveScalarSolver::Preprocessing(CGeometry *geometry, CSolver **solver_c
          unsigned short iMesh, unsigned short iRKStep, unsigned short RunTime_EqSystem, bool Output) {
 
   const bool implicit = (config->GetKind_TimeIntScheme() == EULER_IMPLICIT);
-  const bool muscl = config->GetMUSCL_Scalar();
-  const bool limiter = (config->GetKind_SlopeLimit_Scalar() != NO_LIMITER) &&
-                       (config->GetInnerIter() <= config->GetLimiterIter());
+  const bool muscl    =  config->GetMUSCL_Scalar();
+  const bool limiter  = (config->GetKind_SlopeLimit_Scalar() != NO_LIMITER) &&
+                        (config->GetInnerIter() <= config->GetLimiterIter());
+
+  su2double * scalars = new su2double[nVar];
+
+  for (auto i_point = 0u; i_point < nPoint; i_point++) {
+
+    // Still needed later on to access GetMassDiffusivity()
+    // CFluidModel * fluid_model_local = solver_container[FLOW_SOL]->GetFluidModel();
+
+    for(int iVar = 0; iVar < nVar; iVar++){
+      scalars[iVar] = nodes->GetSolution(i_point, iVar);
+    }
+
+    // Temporary solution: compute mass diffusivity here with variable Cp.
+    std::vector<su2double> SpecificHeatCp;
+    unsigned short n_species_mixture = nVar + 1;
+    SpecificHeatCp.resize(n_species_mixture);
+
+    for (int iVar = 0; iVar < n_species_mixture; iVar++) {
+      SpecificHeatCp.at(iVar) = config->GetSpecific_Heat_Cp(iVar);
+    }
+  
+    su2double cp = SpecificHeatCp[0] * scalars[0] + SpecificHeatCp[1] * (1- scalars[0]);
+    su2double density     = solver_container[FLOW_SOL]->GetNodes()->GetDensity(i_point);
+    su2double thermal_conductivity     = solver_container[FLOW_SOL]->GetNodes()->GetThermalConductivity(i_point);
+    su2double Lewis = 1;
+    su2double mass_diffusivity = thermal_conductivity / (Lewis * density * cp);
+
+    for(auto iVar = 0u; iVar < nVar; iVar++){
+      nodes->SetDiffusivity(i_point, mass_diffusivity, iVar);
+    }
+
+    // for(auto iVar = 0u; iVar < nVar; iVar++){
+    //   nodes->SetDiffusivity(i_point, fluid_model_local->GetMassDiffusivity(), iVar);
+    // }
+
+    if (!Output) LinSysRes.SetBlock_Zero(i_point);
```

Comment:
Why was this added? Not even 10 lines below there is this code which should do the same. please enlighten me
```c++
  if (!ReducerStrategy && !Output) {
    LinSysRes.SetValZero();
    LinSysRes.SetValZero();
    if (implicit) Jacobian.SetValZero();
    if (implicit) Jacobian.SetValZero();
    else {SU2_OMP_BARRIER}
    else {SU2_OMP_BARRIER}
  }
  }
```

Reply:
```suggestion
```